inc/csv/logcsvexport.class.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2020 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Csv;
+
+use CommonDBTM;
+use Log;
+use User;
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+class LogCsvExport implements ExportToCsvInterface
+{
+
+   /** @var CommonDBTM */
+   protected $item;
+
+   /** @var array */
+   protected $filter;
+
+   public function __construct(CommonDBTM $item, array $filter) {
+      $this->item   = $item;
+      $this->filter = $filter;
+   }
+
+   public function getFileName(): string {
+      $name = $this->item->getFriendlyName();
+      $date = date('Y_m_d', time());
+
+      // Avoid whitespace in name
+      $name = str_replace(' ', '_', $name);
+
+      return "{$name}_$date.csv";
```

Comment:
Spaces in filenames is not really more a problem, but if you want to "clean" filenames, you should use `Toolbox::slugify()`.

Also `getFriendlyName()` can return an empty string (name may not be set), it may be better to keep type and id: 

```suggestion
      return Toolbox::slugify("{$this->item->getTypeName(1)}_{$this->item->getId()}_{$name}_$date.csv";
```
Do not forget to add the `use \Toolbox;` directive.

Reply:
hum... `use Toolbox;` no? (no starting backslash)